process_sample_data.py

```diff
@@ -84,16 +85,31 @@ def _parseData(self):
                             .timestamp() * 1000.0)
 
             self._flow_rates.append(float(splitDatum[1]
-                                          .replace(flow_rate_marker,
-                                                   "")
+                                          .replace(flow_rate_marker, "")
                                           .strip("\n")) / 10)
             self._tidal_volumes.append(float(splitDatum[2]
-                                             .replace(tidal_volume_marker,
-                                                      "")
+                                             .replace(tidal_volume_marker, "")
                                              .strip("\n")))
 
             if (pressure_marker in datum):
                 self._pressures.append(float(splitDatum[3]
                                              .replace(pressure_marker,
                                                       "")
                                              .strip("\n")) / 10)
+
+    def _remove_repeat_data(self):
+        non_repeat_indices = []
+        for i in range(len(self._timestamps)):
```

Comment:
Can you populate `non_repeat_indices` as a comprehension as opposed to an append inside a loop?